generators/aws-containers/aws-client.js

```diff
@@ -0,0 +1,299 @@
+        resolve();
+    }));
+}
+
+/**
+ * Retrieve decoded information to authenticate to Docker with AWS credentials.
+ * @returns {Promise} Returns a promise that resolves when the informations are retrieved.
+ */
+function getDockerLogin() {
+    return spinner(new Promise((resolve, reject) => _getAuthorizationToken()
+        .then(authToken => sts.getCallerIdentity({}).promise()
+            .then((data) => {
+                const decoded = utils.decodeBase64(authToken.authorizationToken);
+                const splitResult = _.split(decoded, ':');
+                resolve({
+                    username: splitResult[0],
+                    password: splitResult[1],
+                    accountId: data.Account
+                });
+            })
+            .catch(() => reject(new Error('Couldn\'t retrieve the user informations'))))));
+}
+
+/**
+ * Fetch Authentication token from AWS to authenticate with Docker
+ * @returns {Promise} Returns a promise that resolves when the informations are retrieved.
+ * @private
+ */
+function _getAuthorizationToken() {
+    return spinner(new Promise((resolve, reject) =>
+        ECR.sdk.getAuthorizationToken({}).promise()
+            .then((data) => {
+                if (!_.has(data, 'authorizationData.0')) {
+                    reject(new Error('No authorization data found.'));
+                    return;
+                }
+                resolve(data.authorizationData[0]);
+            })));
+}
+
+/**
+ * Create a S3 Bucket in said region with said bucketBaseName.
+ * the bucketBaseName will be used to create a
+ * @param bucketName the name of the bucket to create.
+ * @param region the region to create the bucket in.
+ * @returns {Promise}
+ */
+function createS3Bucket(bucketName, region = DEFAULT_REGION) {
+    const createBuckerParams = {
+        Bucket: bucketName
+    };
+    return spinner(new Promise((resolve, reject) => s3.headBucket({
+        Bucket: bucketName
+    }).promise()
+        .catch((error) => {
+            if (error.code !== 'NotFound') {
+                reject(new Error(`The S3 Bucket ${chalk.bold(bucketName)} in region ${chalk.bold(region)} already exists and you don't have access to it. Error code: ${chalk.bold(error.code)}`));
+            }
+        })
+        .then(() =>
+            s3.createBucket(createBuckerParams).promise()
+                .then(resolve)
+                .catch(error => reject(new Error(`There was an error during the creation of the S3 Bucket ${chalk.bold(bucketName)} in region ${chalk.bold(region)}`))))));
+}
+
+/**
+ * Upload the template in the S3Bucket
+ * @param bucketName S3 Bucket name to upload the template into
+ * @param filename Name to give to the file in the Bucket
+ * @param path Path to the file
+ * @returns {Promise}
+ */
+function uploadTemplate(bucketName, filename, path) {
+    return spinner(new Promise((resolve, reject) =>
+        fs.stat(path, (error, stats) => {
+            if (_.isNil(stats)) {
```

Comment:
As lodash's `isNil` just does `value == null`, I think you can just write `!stats` here